uniconnect-epfl/uniconnect
# refactor: structure changes and minor fixes

# What I did
This PR mainly covers project structure changes. I also streamlined CI process and updated the README.

# How I did it
1. Refactored project structure to make it more clean and uniform. `Graph` component has been moved into `components` folder. `maestro` end-to-end tests folder has been moved under `__tests__`. Some unused assets and files have been deleted.  Some files have been renamed to follow naming conventions. Type issues have been solved in `RegistrationStackNavigator`. 

  **Important**: all firebase config files and files containing firebase API calls have been refactored under `firebase` folder. This is how we intend to keep it in the future - code logic containing calls to firebase functionalities should be contained in files stored under `firebase`. E.g. `Login.ts` which contains a function that calls the firebase API to authenticate a user.
  
  ![image](https://github.com/uniconnect-epfl/uniconnect/assets/79483254/b8c89917-cc25-4917-aa18-68b1701d8575)

2. Fixed Splash Screen hiding logic to make it more efficient.
3. Fixed some test suites that were failing.
4. Updated the EAS build CI workflow to build an APK each time `main` is updated.
5. Updated the SonarCloud CI to contain a test coverage report. From now on, everyone will need to achieve a certain threshold of 80% test coverage to be able to merge into `main`.
6. Updated README.md to contain useful information about unit testing, end-to-end testing and automatic building.

# How to verify it
- Run the test suite with `npm run test`
- Try to open a PR and check the SonarScan report.
- Adhere to the new project structure.
